### PR TITLE
NIFI-5845: Add support for OTHER and SQLXML JDBC types to SQL/Hive processors

### PR DESCRIPTION
**Pull request by mattyb149**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/util/hive/HiveJdbcCommon.java [hunk 1]
```diff
@@ -402,6 +409,9 @@ public static long convertToCsvStream(final ResultSet rs, final OutputStream out
                             rowValues.add("");
                         }
                         break;
+                    case SQLXML:
+                        rowValues.add(StringEscapeUtils.escapeCsv(((java.sql.SQLXML) value).getString()));
```
Comment (ijokarumawak) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/util/hive/HiveJdbcCommon.java]: Shouldn't we take care of null `value` here, too?

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/util/hive/HiveJdbcCommon.java [hunk 2]
```diff
@@ -402,6 +409,9 @@ public static long convertToCsvStream(final ResultSet rs, final OutputStream out
                             rowValues.add("");
                         }
                         break;
+                    case SQLXML:
+                        rowValues.add(StringEscapeUtils.escapeCsv(((SQLXML) value).getString()));
```
Comment (ijokarumawak) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/util/hive/HiveJdbcCommon.java]: Shouldn't we take care of null `value` here?

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/util/hive/HiveJdbcCommon.java (hunk 1 above)
Comment (mattyb149) [reply nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/util/hive/HiveJdbcCommon.java]: Yep missed that (nulls are handled earlier in the Avro code, but individually in the CSV code), will add both, good catch! I had tested with nulls but only for Avro output.

**Comment by ijokarumawak** [comment]
LGTM +1. Merging. Thanks @mattyb149!

**Pull request closed without merging**